landlab/io/netcdf/read.py

```diff
@@ -247,8 +275,11 @@ def read_netcdf(nc_file, just_grid=False):
     except TypeError:
         root = nc4.Dataset(nc_file, 'r', format='NETCDF4')
 
-    node_coords = _read_netcdf_structured_grid(root)
-
+    try:
+        node_coords = _read_netcdf_structured_grid(root)
+    except:
```

Comment:
No bare `except` statements. Please add a specific exception or, an `if` statement that checks for the grid type.

Maybe something along the lines of,
```python
if is_raster_grid(root):
    node_coords = _read_netcdf_raster_structured_grid(root)
else:
    node_coords = _read_netcdf_structured_grid(root)
```